From 3c667d09475f4ff920f7bfca21c269e01cc5efcf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Tue, 29 Sep 2026 22:57:34 +0200
Subject: [PATCH] avformat/dashdec: copy the init section from the current read
 offset
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Every read after the first copied the init section from its start
again, which corrupted init segments larger than one read.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
---
 libavformat/dashdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index 76352c170831..ffef5bfb2f3a 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -1888,7 +1888,7 @@ restart:
     if (v->init_sec_buf_read_offset < v->init_sec_data_len) {
         /* Push init section out first before first actual fragment */
         int copy_size = FFMIN(v->init_sec_data_len - v->init_sec_buf_read_offset, buf_size);
-        memcpy(buf, v->init_sec_buf, copy_size);
+        memcpy(buf, v->init_sec_buf + v->init_sec_buf_read_offset, copy_size);
         v->init_sec_buf_read_offset += copy_size;
         ret = copy_size;
         goto end;
